File: demo/src/java/eval/KoKoAtomValue.java
/**
 @author loriacarlos@gmail.com
 
*/
package kokoslan.ast;
import java.io.*;

public class KoKoAtomValue<T> implements KoKoValue{
   private T value;
   public T getValue(){return this.value;}
   
   public KoKoAtomValue(T value){
      this.value = value;
   }
   public String toString(){
      return this.value.toString();
   }
}

File: KoKoslan/src/kotlin/eval/KoKoAtomValue.kt
/*
    Andrey Arguedas Espinoza
    Daniela Armas Sanchez
    Michael Chen Wang
    Kimberly Olivas Delgado
*/

package kokoslan.kt.eval
import java.io.*

open class KoKoAtomValue<T>(private var value: T) : KoKoValue{
    
    fun getValue() : T {return value}

    override fun toString(): String{
        return value.toString()
    }
}